sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.maps
+
+import com.mapbox.common.Logger
+import com.mapbox.maps.plugin.delegates.listeners.OnMapLoadErrorListener
+import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+import com.mapbox.maps.plugin.delegates.listeners.eventdata.MapLoadErrorType
+import java.lang.ref.WeakReference
+
+/**
+ * Class that listens to style error and load events
+ * and maintains and invokes user added listeners.
+ */
+internal class StyleObserver(
+  private val mapboxMap: MapboxMap,
+  private val nativeMap: WeakReference<MapInterface>,
+  private val nativeObserver: NativeObserver,
+  private val pixelRatio: Float
+) : OnStyleLoadedListener, OnMapLoadErrorListener {
+
+  private val awaitingStyleLoadListeners = mutableListOf<Style.OnStyleLoaded>()
+  private var awaitingStyleErrorListener: OnMapLoadErrorListener? = null
+
+  init {
+    nativeObserver.addOnStyleLoadedListener(this)
+    nativeObserver.addOnMapLoadErrorListener(this)
+  }
+
+  /**
+   * Clears previous added listeners and setup to receive callback for new style loaded or error events.
+   */
+  fun onNewStyleLoad(loadedListener: Style.OnStyleLoaded?, onMapLoadErrorListener: OnMapLoadErrorListener?) {
+    awaitingStyleLoadListeners.clear()
+    loadedListener?.let {
+      awaitingStyleLoadListeners.add(loadedListener)
+    }
+    awaitingStyleErrorListener = onMapLoadErrorListener
+  }
+
+  /**
+   * Add a style listener invoked when a new style loaded events occurs.
+   */
+  fun addOnStyleLoadListener(loadedListener: Style.OnStyleLoaded) {
```

Comment:
we have add, perhaps we also need a remove?

Reply:
what is the use-case? This is an internal API

Reply:
I see `add` and get triggered to add `remove`. But agree, if internal API - then guess it's good as is.

Reply:
:) 

fwiw, this API cleans up itself either as part of the callback invocation or as part of onDestroy.